python/cudf/cudf/core/_base_index.py

```diff
@@ -1414,6 +1421,22 @@ def from_pandas(cls, index, nan_as_null=None):
     def _constructor_expanddim(self):
         return cudf.MultiIndex
 
+    def _apply_boolean_mask(self, boolean_mask):
+        """Apply boolean mask to each row of `self`.
+
+        Rows corresponding to `False` is dropped.
+        """
+        boolean_mask = cudf.core.column.as_column(boolean_mask)
+        if not is_bool_dtype(boolean_mask.dtype):
+            raise ValueError("boolean_mask is not boolean type.")
+
+        result = self.__class__._from_columns(
```

Comment:
Do we still want to use `_from_columns()` v/s `_from_data`. The following would work too, but now we have more than "One Way To Do It":

```python
result_columns = apply_boolean_mask(list(self._columns), boolean_mask)
result = self.__class__._from_data(
    dict(zip(self._column_names, result_columns))
)
```

Reply:
This question is tangentially related to another review of mine https://github.com/rapidsai/cudf/pull/9750#discussion_r761558418 where I did advocate for having "another way to do it" :). IMO adding new methods that _wrap_ existing methods with some extra wrinkle is not a bad thing since it's explicitly not anti-DRY, but I can definitely see some wiggle room on how much logic needs to be in that wrapper for it to merit being an extra function. I could be convinced either way, but since these are essentially conflicting opinions on two of @isVoid's PRs I think we should align on our plans going forward.

Reply:
I should note that in this case, `_from_columns` currently does some extra handling automatically to split up columns that are index vs data columns, which is out of scope for `_from_data`. That's one good reason to consider `_from_columns` useful even if we decide against the wrapper that I proposed in https://github.com/rapidsai/cudf/pull/9750#discussion_r761558418.

Reply:
We discussed offline and are planning to keep using `_from_columns` for now.